### PR TITLE
Fix vma submodule, ambigous condidate 

### PR DESCRIPTION
**Pull request by Clemapfel**
Fixed submodule not pointing to its intended target.

Because `VK_NULL_HANDLE` is a C-style constant, it is polymorphic (the type is inferred based on usage, not at declaration). Casting it to `vk::Fence` and `vk::PipeLineCache` triggered an "ambiguous candidate" error with G++11 for me. The reason is that `0` can be both interpreted as an int, and a `std::nullptr`, both of whom are valid arguments for different constructors of  `vk::Fence` and `vk::PipeLineCache` respectively.

Don't merge this PR blindly, please first check if it works for you.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -3,9 +3,21 @@ add_library(telescope SHARED telescope.cpp telescope.h)
...
-target_include_directories(telescope PRIVATE ${BULLET_INCLUDE_DIRS})
-target_link_libraries(telescope PRIVATE 
-  SDL2 SDL2_image SDL2_mixer SDL2_net SDL2_ttf 
-  ${BULLET_LIBRARIES}
-  shaderc_shared)
\ No newline at end of file
+target_include_directories(telescope PRIVATE
+    "${BULLET_INCLUDE_DIRS}"
+    "../VulkanMemoryAllocator-Hpp"
+    #/home/clem/Desktop/shaderc/include
```
Comment (Clemapfel) [inline src/CMakeLists.txt:12]: c.f. above

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -3,9 +3,21 @@ add_library(telescope SHARED telescope.cpp telescope.h)
 target_include_directories(telescope PUBLIC .)
 
 find_package(Bullet REQUIRED)
+find_package(SDL2 REQUIRED)
+find_library(shaderc_shared REQUIRED) # NAMES libshaderc_shared.so PATHS /home/clem/Desktop/shaderc/lib)
```
[the remark this answers is not in the document]
Comment (jhigginbotham64) [reply src/CMakeLists.txt:7]: shaderc_shared is part of the Vulkan SDK, which is required for Telescope development. Sorry I didn't give you a heads-up.

[the remark this answers is not in the document]
Comment (jhigginbotham64) [reply src/CMakeLists.txt:7]: Strangely, `find_library` fails for me but including `shaderc_shared` in the call to `target_link_libraries` is fine.

File: .gitmodules [hunk 3]
```diff
@@ -1,3 +1,3 @@
-[submodule "vma"]
-	path = vma
-	url = https://github.com/EIP-Pivot/VulkanMemoryAllocator-Hpp.git
\ No newline at end of file
+[submodule "VulkanMemoryAllocator-Hpp"]
+	path = VulkanMemoryAllocator-Hpp
+	url = https://github.com/EIP-Pivot/VulkanMemoryAllocator-Hpp.git
```
Comment (jhigginbotham64) [inline .gitmodules:3]: The EIP-Pivot fork is no longer the most up-to-date VMA-hpp. I have the latest Vulkan version, and it fails to build on my machine. See here: https://github.com/YaaZ/VulkanMemoryAllocator-Hpp/issues/1

I'll need to change it to the YaaZ fork, which is now "official".

**Comment by jhigginbotham64** [comment]
@Clemapfel I got it to build after a few more changes. How do you want to handle the shaderc issue?

**Comment by jhigginbotham64** [comment]
I'm still trying to fix my dev environment, something crashes whenever I try running Starlight's unit tests, otherwise I'd be more confident in the changes made here.

**Pull request merged**